ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    num_fc_layers: Optional[Optional[int]] = schema_utils.PositiveInteger(
-        default=1,
-        description="The number of stacked fully connected layers.",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["num_fc_layers"],
+    model_variant: Optional[int] = schema_utils.IntegerOptions(
+        [121, 161, 169, 201],
+        default=121,
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVDenseNetEncoder"]["model_variant"],
     )
 
-    fc_layers: Optional[Optional[List[Dict]]] = schema_utils.DictList(
-        default=None,
-        description="A list of dictionaries containing the parameters of all the fully connected layers. The length "
-        "of the list determines the number of stacked fully connected layers and the content of each "
-        "dictionary determines the parameters for a specific layer. The available parameters for each "
-        "layer are: activation, dropout, norm, norm_params, output_size, use_bias, bias_initializer and "
-        "weights_initializer. If any of those values is missing from the dictionary, the default one "
-        "specified as a parameter of the encoder will be used instead. ",
-        parameter_metadata=ENCODER_METADATA["ResNetEncoder"]["fc_layers"],
+
+@register_encoder_config("efficientnet_torch", IMAGE)
+@dataclass
+class TVEfficientNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["efficientnet_torch"],
+        default="efficientnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        [
+            "b0",
+            "b1",
+            "b2",
+            "b3",
+            "b4",
+            "b5",
+            "b6",
+            "b7",
+            "v2_",
+            "v2_m",
+            "v2_l",
+        ],
+        default="b0",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVEfficientNetEncoder"]["model_variant"],
+    )
 
-@register_encoder_config("mlp_mixer", IMAGE)
-@dataclass(repr=False)
-class MLPMixerEncoderConfig(BaseEncoderConfig):
 
+@register_encoder_config("googlenet_torch", IMAGE)
+@dataclass
+class TVGoogLeNetEncoderConfig(TVBaseEncoderConfig):
     type: str = schema_utils.StringOptions(
-        ["mlp_mixer"],
-        default="mlp_mixer",
+        ["googlenet_torch"],
+        default="googlenet_torch",
         allow_none=False,
         description="Type of encoder.",
     )
 
-    dropout: float = schema_utils.FloatRange(
-        default=0.0,
-        min=0,
-        max=1,
-        description="Dropout rate.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["dropout"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional